bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/async/BulkMutation.java

```diff
@@ -441,8 +441,18 @@ public BulkMutation(
                   @Override
                   public void run() {
                     synchronized (BulkMutation.this) {
-                      scheduledFlush = null;
-                      sendUnsent();
+                      // This run method is not synchronized with BulkMutation.this. It could get
+                      // cancelled from line 419, after entering this method and before entering
+                      // this synchronized block. In this case, this thread
+                      // won't get interrupted until later when it tries to acquire resources from
+                      // ResourceLimiter, and all the entries in the currentBatch will fail because
+                      // of InterruptedException.
+                      // Check and make sure the current thread is not interrupted to avoid this
+                      // race condition.
```

Comment:
I think this comment is really useful, but please rephrase a bit to avoid mentioning line numbers because it will almost definitely get out of sync.